app/main.py

```diff
@@ -1,2 +1,113 @@
 class Dictionary:
-    pass
+    LOAD_FACTOR = 2 / 3
+    RESIZE = 2
+    DEFAULT_CAPACITY = 8
+
+    def __init__(self):
+        self.capacity = 8
+        self.storage = [[] for _ in range(self.DEFAULT_CAPACITY)]
+        self.length = 0
+
+    def resize(self):
+        saved_storage = self.storage[:]
+        self.length = 0
+        self.capacity *= self.RESIZE
+        self.storage = [[] for _ in range(self.capacity)]
+
+        for item in saved_storage:
+            if len(item) != 0:
+                self.__setitem__(item[0], item[1])
+
+    def __setitem__(self, key, value):
+        if self.length == int(self.capacity * self.LOAD_FACTOR):
+            self.resize()
+
+        hashed_value = hash(key)
+        index = hashed_value % self.capacity
+
+        while True:
+            if not self.storage[index]:
+                self.storage[index] = [key, value, hashed_value]
+                self.length += 1
+                return
+            elif hashed_value == self.storage[index][2] and \
+                    key == self.storage[index][0]:
+                if self.storage[index][2] is True:
+                    self.length += 1
+                else:
+                    self.storage[index][1] = value
+                return
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key):
+        hashed_value = hash(key)
+        index = hashed_value % self.capacity
+
+        for _ in range(self.capacity):
+            if self.storage[index][0] == \
```

Comment:
```
d = Dictionary()
print(d[0])
```
IndexError: list index out of range instead of KeyError

Reply:
```
@pytest.mark.timeout(5)
def test_deletion():
    items = [(f"Element {i}", i) for i in range(1000)]
    dictionary = Dictionary()
    for key, value in items:
        dictionary[key] = value
    for key, value in items:
        assert dictionary[key] == value
    assert len(dictionary) == len(items)
    for key, value in items:
        del dictionary[key]
    print(len(dictionary))
```

Reply:
The test above produces a KeyError

Reply:
I've fixed